### PR TITLE
AISDK-26: Include version string in user agent

### PR DESCRIPTION
**Pull request by kbridbur**

### CODE REVIEW

File: src/rev_ai/apiclient.py [hunk 1]
```diff
@@ -98,9 +98,14 @@ def submit_job_local_file(
         if callback_url:
             payload['callback_url'] = callback_url
 
+        illegal_chars = ['\\','~','/']
```
[the remark this answers is not in the document]
Comment (kbridbur) [reply src/rev_ai/apiclient.py]: I don't think we should. This is purely for filename upload and I don't think that is necessary or helpful from a user standpoint

File: src/rev_ai/apiclient.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 import requests
 import json
+import re
```
Comment (JDongian) [inline src/rev_ai/apiclient.py]: This isn't necessary. You should be able to just `from . import __version__`.

File: src/rev_ai/apiclient.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JDongian) [reply src/rev_ai/apiclient.py]: I can agree with that. Expanding path to abspath is still nice tho

**Comment by tparnell** [comment]
You may also want to screen for bidirectional characters.  S3 can deal with them, but the .NET SDK cannot.  See FOX-20165 and search for `bidi` in Jira.

**Comment by kbridbur** [comment]
Thanks @tparnell. I'm reducing the scope of this issue because I think handling this should be the responsibility of the API itself. I'll be filing an issue to handle this on our end

File: tests/test_apiclient.py [hunk 3]
```diff
@@ -12,7 +13,7 @@ def test_constructor_with_success(self):
         client = RevAiAPIClient(token)
 
         headers = client.session.headers
-        assert headers.get('User-Agent') == 'python_sdk'
+        assert headers.get('User-Agent') == 'python_sdk-{}'.format(__version__)
```
Comment (amikofalvy) [inline tests/test_apiclient.py]: I think this test should use string literals.
```suggestion
        # Setup
        __version__ = 'test-version'
        
        # Assert
        assert headers.get('User-Agent') == 'python_sdk-test-version'
```

Comment (amikofalvy) [reply tests/test_apiclient.py]: Currently, the test just repeats the implementation code.

File: src/rev_ai/apiclient.py [hunk 4]
```diff
@@ -40,7 +41,7 @@ def __init__(self, access_token):
         self.session = requests.Session()
         self.session.headers.update({
             'Authorization': 'Bearer {}'.format(access_token),
-            'User-Agent': 'python_sdk'
+            'User-Agent': 'python_sdk-{}'.format(__version__)
```
Comment (alexsku) [inline src/rev_ai/apiclient.py]: are f strings available for us here?

Comment (kbridbur) [reply src/rev_ai/apiclient.py]: No, f strings were introduced in python 3.6 to my knowledge.

Comment (alexsku) [reply src/rev_ai/apiclient.py]: i was afraid so :(

File: tests/test_apiclient.py (hunk 3 above)
Comment (alexsku) [reply tests/test_apiclient.py]: @amikofalvy do you mean we should check that the user-agent string is `'python_sdk-{}'.format(__version__)` versus `python_sdk-test-version'`? I personally would have put it as Kyle did it i think but i don't have strong opinion on this

Comment (kbridbur) [reply tests/test_apiclient.py]: Talked with @amikofalvy. We ended up agreeing on this implementation

**Pull request merged**